### PR TITLE
#24964 Removing variantId value from contentlet's JSON field

### LINKED ISSUE #24964: Experiments: Change the way variants are read/written
### Parent Issue

https://github.com/dotCMS/core/issues/22222

### Problem Statement

The logic in `ESContentFactoryImpl#findAllVersions` was modified to call the method `ESContentFactoryImpl#getContantletInodesFromJsonField`, but the thing is that it fails to return all contentlet versions if the `contentlet_as_json` column is null. For this reason, the `ESContentFactoryImpl#getContantletInodesFromJsonField`should be refactored to avoid using the `contentlet_as_json` column to get variants. The suggested changes are: 
- [x] Create a new column in the `contentlet` table to store/read the `variant` field
- [x] Set a default variant for this new column to avoid `null`values
- [x] Rename get**Contantlet**InodesFromJsonField as it has a typo
- [x] Remove `variantId` from com/dotcms/content/model/Contentlet.java

This change implies the creation of an upgrade task and the modification of the postgres.sql file.

### Steps to Reproduce

This issues came up when testing this card https://github.com/dotCMS/core/issues/24093#issuecomment-1544611454

Basically, when a user comes from an upgrade and tries to list the versions of an existing piece of content (tab on the Edit Content functionality), it should fail to show them all.

### Acceptance Criteria

- [ ] Tests must be implemented for the upgrade task
- [ ] Contentlet versions should be listed correctly when the `contentlet_as_json` and/or `variant` columns are set or not

### dotCMS Version

23.06

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* We don't need to include the variantId here anymore, 

https://github.com/dotCMS/core/pull/25035/files#diff-4c3a157c1ca3688be7680e3b7eefa73208cc44d04de7d280d335427c6f300c39L287

this map is going to be Serialize to JSON and save inside the contentlet_as_json.

* Remove variantId method from Contentlet interface

https://github.com/dotCMS/core/pull/25035/files#diff-8ff85036abfa20136fa1fb6ef727e57ed9d05a10338225f75ea92513a5fefa67L57

* Incluse variant_id to save contentlet

https://github.com/dotCMS/core/pull/25035/files#diff-b5d82de048877f797337b47640a36eadbde435004223134e7301ee64ebe9ededR160

*  Rename getContantletInodesFromJsonField as it has a typo and change its implementation to use the new column

https://github.com/dotCMS/core/pull/25035/files#diff-b5d82de048877f797337b47640a36eadbde435004223134e7301ee64ebe9ededR1010

https://github.com/dotCMS/core/pull/25035/files#diff-30db473047bf01b113d2ea4e19f591cd5d08cc41516f01e5bb1ec5be9131dfd7R739

* The variantId need to be set outside the properties Map now

https://github.com/dotCMS/core/pull/25035/files#diff-1367236d40a612b923ececfa521053143f026f60f035c9f9963aae4efdfaf034R182-R1626

https://github.com/dotCMS/core/pull/25035/files#diff-fa1ceaa19618a6b2bbc30e24c6f930b4971f417db50babb748c2e2837ba9eb82R6647

* TU to create the new filed and remove the variantId from the contentlet_as_json field

https://github.com/dotCMS/core/pull/25035/files#diff-8d78df1d2905f3e9ab034c1d88dfdca7ce5d3c7bd13c24b400d1a7116dc1b906R16

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java [hunk 1]
```diff
@@ -2415,4 +2420,51 @@ public void test_getUrlMapForContentlet_with_detail_page_and_pattern() throws Ex
         assertEquals("/testing/urlMapValue1/urlMapValue2",contentletAPI.getUrlMapForContentlet(content, user, false));
     }
 
+    /***
+     * Method to test:
+     * When: Save a {@link Contentlet}
+     * Should: Save the Variant name inside a new column and not inside the Jsong Field
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java:2426]: JSON*

**Review by fabrizzio-dotCMS — APPROVED** [review]
great job. looks really good!

**Review by victoralfaro-dotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        final ArrayList<Map<String, Object>> contentlets_2 = dotConnect.setSQL(
+                        "SELECT contentlet_as_json FROM contentlet WHERE inode = ?")
+                .addParam(contentlet_2.getInode())
+                .loadResults();
+
+        assertEquals(1, contentlets_2.size());
+
+        final Object contentletAsJson = contentlets_2.get(0).get("contentlet_as_json");
+        assertFalse(contentletAsJson != null && UtilMethods.isSet(contentletAsJson.toString()));
+    }
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java:117]: I think an assert for the actual value of the `variant_id` column is missing

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java [hunk 3]
```diff
@@ -1612,11 +1615,19 @@ public ManifestInfo getManifestInfo() {
 	}
 
     public void setVariantId(final String variantId) {
-		map.put(VARIANT_ID, variantId);
+		if (UtilMethods.isSet(variantId)) {
+			this.variantId = variantId;
+		} else {
+			this.variantId = VariantAPI.DEFAULT_VARIANT.name();
+		}
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java:1622]: this.variantId = UtilMethods.isSet(variantId) ? variantId : VariantAPI.DEFAULT_VARIANT.name();

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentlet.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import com.dotmarketing.startup.AbstractJDBCStartupTask;
+import com.dotmarketing.startup.StartupTask;
+import com.liferay.util.StringPool;
+import java.sql.SQLException;
+import java.util.stream.Collectors;
+
+/**
+ * Task to create the variant_id column in the contentlet table, Also it remove the variantId attribute
+ * from the contentlet_as_json column and set the variant_id column to the default value 'DEFAULT'
+ */
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentlet.java:19]: it removes*
sets*

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+            if (columnName.equals("variant_id")) {
+                assertEquals(columnType, "varchar");
+                existsVariantId = true;
+            }
+        }
+
+        assertTrue(existsVariantId);
+    }
+
+    @Test
```
Comment (nollymar) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java:58]: I would add a javadoc or some comments on the test to help understand was it's intended for

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java]: done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R2426

Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java]: done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-093de07a53a725b587963fd60c024f52e089c4f78e7ea88ba8e3a3a0cdaecab9R2426

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java:128]: done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-cf4d6acc2cfbfbdb63e920244db08e5d1193baaf541f4f9eb3ecedbc3b75147eR107-R124

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java]: done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-1367236d40a612b923ececfa521053143f026f60f035c9f9963aae4efdfaf034R1618

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentlet.java (hunk 4 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentlet.java:19]: done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-8d78df1d2905f3e9ab034c1d88dfdca7ce5d3c7bd13c24b400d1a7116dc1b906R17

**Pull request merged**